server/BusinessLogic/ProductManagement/DTOs/CreateProductDTO.cs

```
@@ -0,0 +1,13 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace CoffeeShop.BusinessLogic.ProductManagement.DTOs;
+
+public sealed record CreateProductDTO(
+    string Name,
+    string Description,
+    [RegularExpression(@"^\d+(\.\d{1,2})?$", ErrorMessage = "Price can have up to two decimal places.")]
+    decimal Price,  
+    string ImageUrl,
+    [Range(0, int.MaxValue, ErrorMessage = "Stock must be a non-negative number.")]
```

I don't think we need to type check here, maybe the service layer could handle that?

---

Yes and we should throw `InvalidDomainValueException` for validation to be consistent